### DISCUSSION TITLE
How to convert wav to mp3 using NAudio?

### DISCUSSION THREAD

**Discussion by shruti8789** [naudio/NAudio#941] [General]
I am able to conver mp3 to wav, but for wav to mp3 I just find NAudio.Lame examples online.
Can someone please help me on how to achieve this using NAudio?

**Comment by markheath** [comment]
You can do this with `MediaFoundationEncoder` - take a look [here](https://github.com/naudio/NAudio/blob/master/Docs/MediaFoundationEncoder.md)

**Comment by shruti8789** [comment]
@markheath ,
Thanks for your inputs.
I am able to convert wav to mp3.
I am using the following Wav file config: Sample rate - 44.1KHz, BitsPerSample - 16, channel - 1

When i convert to mp3, i pass the desired bit rate as 192000 (which i see is anyways default in NAudio.Wave.MediaFoundationEncoder),
but the mp3 is returned with bit rate as 128kbps.

**Is this an NAudio limitation? As I am able to convert the same file to mp3 using online tools/ffmpeg.**

one more data point: It is able to convert when the wav file is 2 channel.

My code:
using (WaveFileReader waveFileReader = new WaveFileReader(inputStream))
{
    var mediaType = MediaFoundationEncoder.SelectMediaType(AudioSubtypes.MFAudioFormat_MP3, 
                                                                                             waveFileReader.WaveFormat, desiredBitRate);
    using (var encoder = new MediaFoundationEncoder(mediaType))
    {
        MediaFoundationApi.Startup();
        MediaFoundationEncoder.EncodeToMp3(waveFileReader, outputFilePath, desiredBitRate);
        MediaFoundationApi.Shutdown();
    }
}

**Reply by shruti8789** [reply]
@markheath a gentle reminder here
